packages/vscode-extension/src/webview/views/LaunchConfigurationView.tsx

```diff
@@ -0,0 +1,267 @@
+import "./View.css";
+import "./LaunchConfigurationView.css";
+import Label from "../components/shared/Label";
+import { useLaunchConfig } from "../providers/LaunchConfigProvider";
+import { LaunchConfigurationOptions } from "../../common/LaunchConfig";
+import Select from "../components/shared/Select";
+import { useRef } from "react";
+import { valid } from "semver";
+
+function LaunchConfigurationView() {
+  const { android, appRoot, ios, isExpo, metroConfigPath, env, update, xcodeSchemes } =
+    useLaunchConfig();
+
+  return (
+    <>
+      <Label>iOS</Label>
+      <IosConfiguration
+        scheme={ios?.scheme}
+        configuration={ios?.configuration}
+        update={update}
+        xcodeSchemes={xcodeSchemes}
+      />
+      <div className="launch-configuration-section-margin" />
+
+      <Label>Android</Label>
+      <AndroidConfiguration
+        buildType={android?.buildType}
+        productFlavor={android?.productFlavor}
+        update={update}
+      />
+
+      <div className="launch-configuration-section-margin" />
+
+      <Label>App Root</Label>
+      <AppRootConfiguration appRoot={appRoot} update={update} />
+      <div className="launch-configuration-section-margin" />
+
+      <Label>metro Config Path</Label>
+      <MetroConfigPathConfiguration metroConfigPath={metroConfigPath} update={update} />
+
+      <div className="launch-configuration-section-margin" />
+
+      <Label>is Expo</Label>
+      <IsExpoConfiguration isExpo={isExpo} update={update} />
+
+      <div className="launch-configuration-section-margin" />
+    </>
+  );
+}
+
+interface iosConfigurationProps {
+  scheme?: string;
+  configuration?: string;
+  update: <K extends keyof LaunchConfigurationOptions>(
+    key: K,
+    value: LaunchConfigurationOptions[K]
+  ) => void;
+  xcodeSchemes: string[];
+}
+
+function IosConfiguration({ scheme, configuration, update, xcodeSchemes }: iosConfigurationProps) {
+  const configurationInputRef = useRef<HTMLInputElement>(null);
+
+  const onSchemeChange = (newScheme: string) => {
+    if (newScheme === "Auto") {
+      update("ios", { scheme: undefined, configuration });
+      return;
+    }
+    update("ios", { scheme: newScheme, configuration });
```

Comment:
Nit: My opinion but I dislike using returns like that. I'd write this code as:

```
let scheme: string | undefined = newScheme;
if(newScheme === "Auto") {
  scheme = undefined;
}
update("ios", { scheme: newScheme, configuration });
```

so it doesn't have multiple points of doing the same side-effect and we only manipulate the input.